### PR TITLE
don't break older returners right now

### LINKED ISSUE #48306: Documentation update for custom returners used for master job cache
### Description of Issue/Question

The [documentation for master job cache support for returners](https://docs.saltstack.com/en/latest/ref/returners/#master-job-cache-support) needs a couple updates:

1. The [salt.utils.jid.gen_jid](https://github.com/saltstack/salt/blob/v2018.3.1/salt/utils/jid.py#L17) function now requires the salt config object (e.g. `__opt__`) as an argument.
1. The `save_load` function [now requires a third argument](https://github.com/saltstack/salt/blob/v2018.3.1/salt/master.py#L2216): `minions`.

### Setup

N/A

### Steps to Reproduce Issue

N/A

### Versions Report

```
           Salt: 2018.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.3.7
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Nov 24 2017, 17:33:09)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1

System Versions:
           dist: debian 9.4
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-6-amd64
         system: Linux
        version: debian 9.4
```

### PR DESCRIPTION
**Pull request by gtmanfred**
### What issues does this PR fix or reference?
Fixes #48306 

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by DmitryKuzmenko — APPROVED** [review]

File: salt/master.py [hunk 1]
```diff
@@ -2211,7 +2211,11 @@ def _prep_pub(self, minions, jid, clear_load, extra, missing):
 
             if save_load_func:
                 try:
-                    self.mminion.returners[fstr](clear_load['jid'], clear_load, minions=minions)
+                    returner_args = [clear_load['jid'], clear_load]
```
Comment (cachedout) [inline salt/master.py]: IMHO, this whole addition needs some inline documentation to explain the purpose of this code.

**Comment by rallytime** [comment]
@gtmanfred In addition to the doc request above, looks like lint failed here as well.

File: salt/master.py (hunk 1 above)
Comment (terminalmage) [reply salt/master.py]: Yes, this looks good but documentation would be helpful for future maintenance.

**Review by terminalmage — CHANGES_REQUESTED** [review]

File: salt/master.py (hunk 1 above)
Comment (gtmanfred) [reply salt/master.py]: Looks like this was added back in 2015.8, but was just never documented for save_load in the returner, I am going to revert this change, and just add documentation about it.

**Review by rallytime — APPROVED** [review]

**Review by terminalmage — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Pull request merged**